### ISSUE TITLE
request for help to encrypt a file

### ISSUE THREAD

**Issue by rowrys** [lasere77/loggepy#2]
Hello, it's been several days that I try to make a Script in python that would encrypt a file but I can't manage to do it ... if someone could help me it would be very nice thank you

**Comment by ghost** [comment]
I think this will help

    import rsa

    def encrypt(data, pub):
        return rsa.encrypt(data.encode(), pub)

    def decrypt(data, priv):
        return rsa.decrypt(eData, priv)


    if __name__ == "__main__":

        f = open("data.txt", "r")
        data = f.read()

        pub, priv = rsa.newkeys(512)

        eData = encrypt(data, pub)
        print(f"\nEncrypted data: {eData}\n")

        dData = decrypt(data, priv)
        print(f"Decrypted data: {dData}\n")

**Comment by rowrys** [comment]
hello thank you very much for the help !!!!!! I will try to put it in place very quickly. I will give news here if it ever works or not . just a small question that has no connection how did you manage to find my project (I don't really know how it works😅)?

**Comment by rowrys** [comment]
`
import rsa

def encrypt(data, pub):
    return rsa.encrypt(data.encode(), pub)

def decrypt(data, priv):
    return rsa.decrypt(eData, priv)

if __name__ == '__main__':

    f = open(r"test.txt", "r")
    data = f.read()
    f.close()

    pub, priv = rsa.newkeys(512)

    eData = encrypt(data, pub)
    print(eData)
    
    f = open(r"test.txt", "w")
    dData = decrypt(data, priv)
    print(dData)
    f.write(dData)
    f.close()
`

hello, I can test the Script that you pass to me but I have them a small error 😅 infact when I try to encrypt what I have just encrypted I have this error "OverflowError: 126 bytes needed for message, but there is only space for 31" 

so to sum up when I encrypt the file for example if I write 'test' it will encrypt it well and write it in the file in this form "b'4>h\xbeZ\xaa\xcd\x0c\xb6I\x89\ xde\x7f,U\x8a]\x8b\xc7~\xfd\xf75\xc9\x7f\x03\xcb\x99 \x99E\\\xf6\x05\xb6M\x12\x91\xa6b\xd9\x8d\x9b\ xef=D^\xcde2\xab\xee\xcc`\x9c\x86\t\xa0\xcc\x81f\xe0\xd1\x1c'" and so when I go to decipher (with the code that is just above) it returns me the so-called plus rate error . thanks for the help !!!

(sorry for the code he does not want to put it to me in full ...)

**Comment by ghost** [comment]
> will give news here if it ever works or not . just a small question that has no connection how did you manage to find my project (I d

I found it out just by searching for the tag "help wanted"

**Comment by rowrys** [comment]
OK, thanks

**Comment by ghost** [comment]
This worked for me, try this:
    import rsa

    def encrypt(data, pub):
        return rsa.encrypt(data.encode(), pub)

    def decrypt(data, priv):
        return rsa.decrypt(eData, priv)


    if __name__ == "__main__":

        f = open("test.txt", "rb")
        data = f.read()
        f.close()

        # Generate keys
        # You MUST save the keys, Priv key to decrypt, Pub to decrypt data
        pub, priv = rsa.newkeys(2048)

        eData = encrypt(data, pub)
        print(f"[DEBUG] Encrypted data: {eData}")

        print("\n\n")

        f = open(r"test.txt", "wb")
        dData = decrypt(data, priv)
        print(f"[DEBUG] Decrypted data: {dData}")
        f.write(dData) 
        f.close()

**Comment by ghost** [comment]
Tell me if this works now

**Comment by ghost** [comment]
priv key to decrypt and pub key to encrypt*

**Comment by ghost** [comment]
Just to remember, this script generate new keys each test. If you try to decrypt data from previous tests it won't work, the key are not the same. This script is just a sample and won't save or use saved keys. If you need help to save and use the keys, tell me

**Comment by rowrys** [comment]
thank you very much for your answer I tried the second Script that you send me but I have an error ... I am really bad ... I think the error comes from the lines where you open the file more precisely open mode l "b" crashes the script with this error "Traceback (most recent call last):
  File "c:\Users\laser\Desktop\fichier o le bg\ho le bg .py", line 20, in <module>
    eData = encrypt(data, pub)
  File "c:\Users\laser\Desktop\fichier o le bg\ho le bg .py", line 4, in encrypt
    return rsa.encrypt(data.encode(), pub)
AttributeError: 'bytes' object has no attribute 'encode' " .
 I will try to change it tomorrow , sorry for all this is a question .

**Comment by ghost** [comment]
Ah, I forgot to update that, just remove the ".encode()" from the encrypt function

**Comment by ghost** [comment]
I found out what causes the overflow: 
>"As a rule of thumb, you can only encrypt data as large as the RSA key length. So, if you've got a 4096-bit RSA key, you can only encrypt messages up to 4096 bits long." (https://security.stackexchange.com/questions/33434/rsa-maximum-bytes-to-encrypt-comparison-to-aes-in-terms-of-security)

**Comment by rowrys** [comment]
ok I tested it but its still not working it still gives me the same error😕

**Comment by rowrys** [comment]
I finally found a libreri that fits what I want to do thanks.
(cryptographie)
